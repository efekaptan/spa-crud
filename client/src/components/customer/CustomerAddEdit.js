import React from 'react';
import Calendar from '../common/Calendar';

const CustomerAddEdit = ({ customer, onChange, onSubmit, onGoBack }) =>
    <div className="row">
        <div className="col-lg-6">
            <div className="card">
                <form className="form-horizontal">
                    <div className="card-header">
                        <strong>{customer.id ? "Edit" : "Add"} Customer</strong>
                    </div>
                    <div className="card-body card-block">
                        <div className="row form-group">
                            <div className="col col-md-3">
                                <label htmlFor="firstName" className="form-control-label">First Name</label>
                            </div>
                            <div className="col-12 col-md-9">
                                <input type="text" id="firstName" name="firstName" value={customer.firstName} onChange={onChange} placeholder="Enter First Name..." className="form-control" />
                            </div>
                        </div>
                        <div className="row form-group">
                            <div className="col col-md-3">
                                <label htmlFor="lastName" className="form-control-label">Last Name</label>
                            </div>
                            <div className="col-12 col-md-9">
                                <input type="text" id="lastName" name="lastName" value={customer.lastName} onChange={onChange} placeholder="Enter Last Name..." className="form-control" />
                            </div>
                        </div>
                        <div className="row form-group">
                            <div className="col col-md-3">
                                <label htmlFor="birthday" className="form-control-label">Birthday</label>
                            </div>
                            <div className="col-12 col-md-9">
                                <Calendar onChange={(field, value) => onChange({ target: { name: 'birthday', value: value } })} value={customer.birthday} />
                            </div>
                        </div>
                        <div className="row form-group">
                            <div className="col col-md-3">
                                <label htmlFor="gender" className="form-control-label">Gender</label>
                            </div>
                            <div className="col-12 col-md-9">
                                <div className="radio-inline">
                                    <input className="radio-input" type="radio" name="gender" id="inlineRadio1" value="m" checked={customer.gender === "m"} onChange={onChange} />
                                    <label className="radio-label" htmlFor="inlineRadio1">Male</label>
                                </div>
                                <div className="radio-inline">
                                    <input className="radio-input" type="radio" name="gender" id="inlineRadio2" value="w" checked={customer.gender === "w"} onChange={onChange} />
                                    <label className="radio-label" htmlFor="inlineRadio2">Female</label>
                                </div>
                            </div>
                        </div>
                        <div className="row form-group">
                            <div className="col col-md-3">
                                <label htmlFor="lastContact" className="form-control-label">Last Contact</label>
                            </div>
                            <div className="col-12 col-md-9">
                                <Calendar onChange={(field, value) => onChange({ target: { name: 'lastContact', value: value } })} value={customer.lastContact} />
                            </div>
                        </div>
                        <div className="row form-group">
                            <div className="col col-md-3">
                                <label htmlFor="customerLifetimeValue" className="form-control-label">Lifetime</label>
                            </div>
                            <div className="col-12 col-md-9">
                                <input type="text" id="customerLifetimeValue" name="customerLifetimeValue" value={customer.customerLifetimeValue} onChange={onChange} placeholder="Enter Customer Lifetime Value..." className="form-control" />
                            </div>
                        </div>
                    </div>
                    <div className="card-footer buttons">
                        <button type="button" onClick={onSubmit} className="btn btn-primary btn-sm">
                            <i className="fa fa-dot-circle-o"></i> {customer.id ? "Save Changes" : "Create"}
                        </button>
                        <button type="button" className="btn btn-sm" onClick={onGoBack}>Go Back</button>
                    </div>
                </form>
            </div>
        </div>
    </div>

export default CustomerAddEdit;