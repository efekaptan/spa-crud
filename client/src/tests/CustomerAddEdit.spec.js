import React from 'react'
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import CustomerAddEdit from '../components/customer/CustomerAddEdit';
import getContextProvider, { contextObject } from './ContextProvider';

configure({ adapter: new Adapter() });

const mockCustomer = {
    id: 15,
    firstName: "Eliseo",
    lastName: "Goldfarb",
    birthday: "1969-11-19T21:00:00.000Z",
    gender: "m",
    lastContact: "2017-03-18T12:20:06.702Z",
    customerLifetimeValue: 1019.91
};

describe('Customer Add/Edit Snapshot', () => {
    const reactElement = getContextProvider(contextObject,
        <CustomerAddEdit customer={mockCustomer} onChange={()=>{}} />
    );
    it('Capturing Snapshot of Customer Add/Edit Snapshot', () => {
        const renderedValue = renderer.create(reactElement).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('Component Render Tests', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <MemoryRouter>
                <CustomerAddEdit customer={mockCustomer} onChange={()=>{}} />
            </MemoryRouter>);
    });

    it('Render the DUMB component', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('Display "Eliseo" in first input', () => {
        expect(wrapper.find('#firstName').props().value).toEqual('Eliseo');
    });

    it('Display "Goldfarb" in first input', () => {
        expect(wrapper.find('#lastName').props().value).toEqual('Goldfarb');
    });

    it('Display "20-Nov-1969" in birthday input', () => {
        expect(wrapper.find('.form-control').at(2).props().value).toEqual('20-Nov-1969');
    });

    it('Display "18-Mar-2017" in last contact input', () => {
        expect(wrapper.find('.form-control').at(3).props().value).toEqual('18-Mar-2017');
    });

    it('Display "1019.91" in customerLifetimeValue input', () => {
        expect(wrapper.find('#customerLifetimeValue').props().value).toEqual(1019.91);
    });
});