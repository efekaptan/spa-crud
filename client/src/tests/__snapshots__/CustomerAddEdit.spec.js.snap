// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Add/Edit Snapshot Capturing Snapshot of Customer Add/Edit Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6"
  >
    <div
      className="card"
    >
      <form
        className="form-horizontal"
      >
        <div
          className="card-header"
        >
          <strong>
            Edit
             Customer
          </strong>
        </div>
        <div
          className="card-body card-block"
        >
          <div
            className="row form-group"
          >
            <div
              className="col col-md-3"
            >
              <label
                className="form-control-label"
                htmlFor="firstName"
              >
                First Name
              </label>
            </div>
            <div
              className="col-12 col-md-9"
            >
              <input
                className="form-control"
                id="firstName"
                name="firstName"
                onChange={[Function]}
                placeholder="Enter First Name..."
                type="text"
                value="Eliseo"
              />
            </div>
          </div>
          <div
            className="row form-group"
          >
            <div
              className="col col-md-3"
            >
              <label
                className="form-control-label"
                htmlFor="lastName"
              >
                Last Name
              </label>
            </div>
            <div
              className="col-12 col-md-9"
            >
              <input
                className="form-control"
                id="lastName"
                name="lastName"
                onChange={[Function]}
                placeholder="Enter Last Name..."
                type="text"
                value="Goldfarb"
              />
            </div>
          </div>
          <div
            className="row form-group"
          >
            <div
              className="col col-md-3"
            >
              <label
                className="form-control-label"
                htmlFor="birthday"
              >
                Birthday
              </label>
            </div>
            <div
              className="col-12 col-md-9"
            >
              <input
                className="form-control"
                onBlur={undefined}
                onClick={[Function]}
                onContextMenu={undefined}
                onFocus={undefined}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={undefined}
                onMouseLeave={undefined}
                onTouchStart={[Function]}
                readOnly={true}
                value="20-Nov-1969"
              />
            </div>
          </div>
          <div
            className="row form-group"
          >
            <div
              className="col col-md-3"
            >
              <label
                className="form-control-label"
                htmlFor="gender"
              >
                Gender
              </label>
            </div>
            <div
              className="col-12 col-md-9"
            >
              <div
                className="radio-inline"
              >
                <input
                  checked={true}
                  className="radio-input"
                  id="inlineRadio1"
                  name="gender"
                  onChange={[Function]}
                  type="radio"
                  value="m"
                />
                <label
                  className="radio-label"
                  htmlFor="inlineRadio1"
                >
                  Male
                </label>
              </div>
              <div
                className="radio-inline"
              >
                <input
                  checked={false}
                  className="radio-input"
                  id="inlineRadio2"
                  name="gender"
                  onChange={[Function]}
                  type="radio"
                  value="w"
                />
                <label
                  className="radio-label"
                  htmlFor="inlineRadio2"
                >
                  Female
                </label>
              </div>
            </div>
          </div>
          <div
            className="row form-group"
          >
            <div
              className="col col-md-3"
            >
              <label
                className="form-control-label"
                htmlFor="lastContact"
              >
                Last Contact
              </label>
            </div>
            <div
              className="col-12 col-md-9"
            >
              <input
                className="form-control"
                onBlur={undefined}
                onClick={[Function]}
                onContextMenu={undefined}
                onFocus={undefined}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={undefined}
                onMouseLeave={undefined}
                onTouchStart={[Function]}
                readOnly={true}
                value="18-Mar-2017"
              />
            </div>
          </div>
          <div
            className="row form-group"
          >
            <div
              className="col col-md-3"
            >
              <label
                className="form-control-label"
                htmlFor="customerLifetimeValue"
              >
                Lifetime
              </label>
            </div>
            <div
              className="col-12 col-md-9"
            >
              <input
                className="form-control"
                id="customerLifetimeValue"
                name="customerLifetimeValue"
                onChange={[Function]}
                placeholder="Enter Customer Lifetime Value..."
                type="text"
                value={1019.91}
              />
            </div>
          </div>
        </div>
        <div
          className="card-footer buttons"
        >
          <button
            className="btn btn-primary btn-sm"
            onClick={undefined}
            type="button"
          >
            <i
              className="fa fa-dot-circle-o"
            />
             
            Save Changes
          </button>
          <button
            className="btn btn-sm"
            onClick={undefined}
            type="button"
          >
            Go Back
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
